scripts/reboot

```diff
@@ -341,4 +375,37 @@ if [ -x ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT} ]; then
 fi
 
 VERBOSE=yes debug "Issuing OS-level reboot ..." >&2
-exec /sbin/reboot ${REBOOT_FLAGS}
+
+if [[ "${BLOCKING_MODE}" != "yes" ]]; then
+    # In non blocking mode, we will remain the original logic.
+    exec /sbin/reboot ${REBOOT_FLAGS}
+else
+
+    REBOOT_START_TIME=$(date +%s)
+    /sbin/reboot ${REBOOT_FLAGS}
+    if [[ "${BLOCKING_MODE}" == "yes" ]]; then
+        # In blocking mode, we just deadloop here to avoid more output generated.
+        BLOCKING_TIMER=1
+        sleep 1
+        while true; do
+            if [ "${SHOW_TIMER}" == "yes" ]; then
+                if [ "${BLOCKING_TIMER}" -ge 50 ]; then
+                    BLOCKING_TIMER=1
+                    echo "."
+                else
+                    let BLOCKING_TIMER++
+                    echo -n "."
+                fi
+            fi
+
+            CURRENT_TIME=$(date +%s)
+            ELAPSED=$((CURRENT_TIME - REBOOT_START_TIME))
+            if (( ELAPSED >= BLOCKING_MODE_TIMEOUT_IN_SECOND )); then
+                echo "The reboot used $(ELAPSED)seconds while the timeout is configed as $(BLOCKING_MODE_TIMEOUT_IN_SECOND)seconds"
+                break
```

Comment:
We should define a ERR code for timeout scenario, e.g.,
```bash
declare -r EXIT_TIMEOUT=2
```
Then replace `break` with `exit EXIT_TIMEOUT`